spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -370,14 +371,56 @@ public void alterNamespace(String[] namespace, NamespaceChange... changes) throw
 
   @Override
   public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException {
+    // Spark assumes that catalogs CASCADE by default. So if we have a namespace catalog
+    /// we need to attempt to recursively delete.
     if (asNamespaceCatalog != null) {
+      Namespace asNamespace = Namespace.of(namespace);
```

Comment:
Since we may exit early, I think you should avoid creating this variable until it is necessary.